### PR TITLE
Fail on compiler errors

### PR DESCRIPTION
**Pull request by ejball**
Various changes, but, most importantly, fails on compiler errors (and fixes them). Bradley helped.

### CODE REVIEW

File: tests/Faithlife.Analyzers.Tests/Helpers/DiagnosticVerifier.Helper.cs [hunk 1]
```diff
@@ -16,10 +19,18 @@ namespace Faithlife.Analyzers.Tests
...
+			"System.Collections",
+			"System.Linq",
+			"System.Private.CoreLib",
+			"System.Runtime",
+			"Microsoft.CodeAnalysis",
+			"Microsoft.CodeAnalysis.CSharp",
+		};
+
+		private static readonly IReadOnlyList<MetadataReference> s_metadataReferences = s_assemblyReferences
+			.Select(x => (MetadataReference) MetadataReference.CreateFromFile(Assembly.Load(x).Location)).ToList();
```
Comment (bgrainger) [inline tests/Faithlife.Analyzers.Tests/Helpers/DiagnosticVerifier.Helper.cs]: Nitpick: move these to the end of the class if we're rewriting them anyway.

File: tests/Faithlife.Analyzers.Tests/Helpers/DiagnosticVerifier.Helper.cs [hunk 2]
```diff
@@ -58,7 +69,10 @@ protected static Diagnostic[] GetSortedDiagnosticsFromDocuments(DiagnosticAnalyz
 			var diagnostics = new List<Diagnostic>();
 			foreach (var project in projects)
 			{
-				var compilationWithAnalyzers = project.GetCompilationAsync().Result.WithAnalyzers(ImmutableArray.Create(analyzer));
+				var compilation = project.GetCompilationAsync().Result;
```
Comment (bgrainger) [inline tests/Faithlife.Analyzers.Tests/Helpers/DiagnosticVerifier.Helper.cs]: Pre-existing bug, but should prefer `.GetAwaiter().GetResult()`.

File: tests/Faithlife.Analyzers.Tests/Helpers/DiagnosticVerifier.Helper.cs [hunk 3]
```diff
@@ -155,10 +169,8 @@ private static Project CreateProject(string[] sources, string language = Languag
 			var solution = workspace
 				.CurrentSolution
 				.AddProject(projectId, TestProjectName, TestProjectName, language)
-				.AddMetadataReference(projectId, CorlibReference)
-				.AddMetadataReference(projectId, SystemCoreReference)
-				.AddMetadataReference(projectId, CSharpSymbolsReference)
-				.AddMetadataReference(projectId, CodeAnalysisReference);
+				.WithProjectCompilationOptions(projectId, new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary))
+				.AddMetadataReferences(projectId, s_metadataReferences);
```
Comment (bgrainger) [inline tests/Faithlife.Analyzers.Tests/Helpers/DiagnosticVerifier.Helper.cs]: 👍

File: tests/Faithlife.Analyzers.Tests/Helpers/DiagnosticVerifier.Helper.cs (hunk 1 above)
Comment (ejball) [reply tests/Faithlife.Analyzers.Tests/Helpers/DiagnosticVerifier.Helper.cs]: 1d8982e6c7a50812adcd333ea8cd3a88ca6fdcf7

File: tests/Faithlife.Analyzers.Tests/Helpers/DiagnosticVerifier.Helper.cs (hunk 2 above)
Comment (ejball) [reply tests/Faithlife.Analyzers.Tests/Helpers/DiagnosticVerifier.Helper.cs]: 6504db14b1a934e7ec05c5eeef771e9ea8edd594

**Pull request merged**